packages/server/src/types.ts

```diff
@@ -45,18 +45,21 @@ export type AnyFunctionMap = {
 
 export type Unsubscriber = () => void;
 
-export type GraphQLServiceConfig = {
+// TODO(AS4): rename
+export type GraphQLServiceConfig<TContext extends BaseContext> = {
   schema: GraphQLSchema;
-  executor: GraphQLExecutor | null;
+  executor: GraphQLExecutor<TContext> | null;
 };
 
 export type SchemaLoadOrUpdateCallback = (schemaContext: {
   apiSchema: GraphQLSchema;
   coreSupergraphSdl: string;
 }) => void;
 
-export interface GatewayInterface {
-  load(options: { apollo: ApolloConfig }): Promise<GraphQLServiceConfig>;
+export interface GatewayInterface<TContext extends BaseContext> {
```

Comment:
This change forces an update on gateway's usage of the interface, which is maybe ok? I know we've removed some back-compat with older gateway versions. I'm wondering if there's value in a default `= BaseContext` here, or if that wouldn't work anyway once we're using `in out`.

Reply:
I kinda assume there's going to be a minimum Gateway version required for AS4 anyway?

Reply:
Yeah, I think that's a fine answer to this question. Very reasonable that moving to AS4 requires you be on a latest version of Gateway.